SofaKernel/modules/SofaBaseLinearSolver/src/SofaBaseLinearSolver/CGLinearSolver.inl

```diff
@@ -33,72 +33,60 @@ namespace sofa::component::linearsolver
-    , f_warmStart( initData(&f_warmStart,false,"warmStart","Use previous solution as initial solution") )
-    , f_verbose( initData(&f_verbose,false,"verbose","Dump system state at each iteration") )
-    , f_graph( initData(&f_graph,"graph","Graph of residuals at each iteration") )
+    : d_maxIter( initData(&d_maxIter,(unsigned)25,"iterations","Maximum number of iterations of the Conjugate Gradient solution") )
+    , d_tolerance( initData(&d_tolerance,(SReal)1e-5,"tolerance","Desired accuracy of the Conjugate Gradient solution evaluating: |r|²/|b|² (ratio of current residual norm over initial residual norm)") )
+    , d_smallDenominatorThreshold( initData(&d_smallDenominatorThreshold,(SReal)1e-5,"threshold","Minimum value of the denominator (pT A p)^ in the conjugate Gradient solution") )
+    , d_warmStart( initData(&d_warmStart,false,"warmStart","Use previous solution as initial solution") )
+    , d_graph( initData(&d_graph,"graph","Graph of residuals at each iteration") )
 {
-    f_graph.setWidget("graph");
-    f_maxIter.setRequired(true);
-    f_tolerance.setRequired(true);
-    f_smallDenominatorThreshold.setRequired(true);
+    d_graph.setWidget("graph");
+    d_maxIter.setRequired(true);
+    d_tolerance.setRequired(true);
+    d_smallDenominatorThreshold.setRequired(true);
 }
 
+/// Initialization function checking input Data
 template<class TMatrix, class TVector>
 void CGLinearSolver<TMatrix,TVector>::init()
 {
-    if(f_verbose.getValue())
-    {
-        this->f_printLog.setValue(true);
-    }
-
-    if(f_tolerance.getValue() < 0.0)
+    if(d_tolerance.getValue() < 0.0)
     {
         msg_warning() << "'tolerance' must be a positive value" << msgendl
                       << "default value used: 1e-5";
-        f_tolerance.setValue(1e-5);
+        d_tolerance.setValue(1e-5);
     }
-    if(f_smallDenominatorThreshold.getValue() < 0.0)
+    if(d_smallDenominatorThreshold.getValue() < 0.0)
     {
         msg_warning() << "'threshold' must be a positive value" << msgendl
                       << "default value used: 1e-5";
-        f_smallDenominatorThreshold.setValue(1e-5);
+        d_smallDenominatorThreshold.setValue(1e-5);
     }
 
     timeStepCount = 0;
     equilibriumReached = false;
 }
 
-template<class TMatrix, class TVector>
-void CGLinearSolver<TMatrix,TVector>::reinit()
-{
-    if(f_verbose.getValue())
-    {
-        this->f_printLog.setValue(true);
-    }
-}
-
+/// Clear graph and clean the RHS / LHS vectors
 template<class TMatrix, class TVector>
 void CGLinearSolver<TMatrix,TVector>::resetSystem()
 {
-    f_graph.beginEdit()->clear();
-    f_graph.endEdit();
+    d_graph.beginEdit()->clear();
+    d_graph.endEdit();
 
     Inherit::resetSystem();
 }
 
+/// Pass the coefficients multiplying the matrices M, B and K from the ODE to the LinearSolver (and includes a resetSystem)
```

Comment:
This is the case only for `GraphScattered` types. With other types, the global system matrix is setup, and `addMBKToMatrix` is called.